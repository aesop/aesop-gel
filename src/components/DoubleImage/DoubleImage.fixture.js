export default {
  imageOne: {
    altText:
      'A tactile approach infuses our work, from our product development to our retail service and visual merchandising.',
    caption:
      'Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    large: './assets/images/DoubleImage/image-one-large.jpg',
    medium: './assets/images/DoubleImage/image-one-medium.jpg',
    small: './assets/images/DoubleImage/image-one-small.jpg',
  },
  imageTwo: {
    altText:
      'Image of: Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    caption:
      'Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    large: './assets/images/DoubleImage/image-two-large.jpg',
    medium: './assets/images/DoubleImage/image-two-medium.jpg',
    small: './assets/images/DoubleImage/image-two-small.jpg',
  },
};
