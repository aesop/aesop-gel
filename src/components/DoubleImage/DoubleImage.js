import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Figure from '~/components/Figure';
import Heading from '~/components/Heading';
import Hyperlink from '~/components/Hyperlink';
import Image from '~/components/Image';
import Video from '~/components/Video';
import styles from './DoubleImage.module.css';

const DoubleImage = ({
  caption1,
  caption2,
  className,
  imageVideo1,
  imageVideo2,
  link1,
  link2,
  theme,
  title1,
  title2,
}) => {
  const classSet = cx(styles.base, styles[theme], className);

  const ImageVideoBlock1 = () => {
    return imageVideo1.type === 'video' ? (
      <Video
        hasAllowAudio={false}
        hasAutoplay={true}
        hasControls={false}
        hasLoop={true}
        hasPlayInFullScreen={false}
        isFullWidth={true}
        large={imageVideo1.large}
        medium={imageVideo1.medium}
        poster={imageVideo1.poster}
        small={imageVideo1.small}
      />
    ) : (
      <Image
        altText={imageVideo1.altText}
        large={imageVideo1.large}
        medium={imageVideo1.medium}
        small={imageVideo1.small}
      />
    );
  };

  const ImageVideoBlock2 = () => {
    return imageVideo2.type === 'video' ? (
      <Video
        hasAllowAudio={false}
        hasAutoplay={true}
        hasControls={false}
        hasLoop={true}
        hasPlayInFullScreen={false}
        isFullWidth={true}
        large={imageVideo2.large}
        medium={imageVideo2.medium}
        poster={imageVideo2.poster}
        small={imageVideo2.small}
      />
    ) : (
      <Image
        altText={imageVideo2.altText}
        large={imageVideo2.large}
        medium={imageVideo2.medium}
        small={imageVideo2.small}
      />
    );
  };

  return (
    <div className={classSet}>
      <Figure
        caption={!title1 && !caption1 && imageVideo1.caption}
        className={styles.figure}
        hasCaptionBorder={false}
        theme={theme}
      >
        {link1 ? (
          <Hyperlink {...link1} className={styles.imageLinkWrapper}>
            <ImageVideoBlock1 />
          </Hyperlink>
        ) : (
          <ImageVideoBlock1 />
        )}
        {(title1 || caption1) && (
          <figcaption className={styles.figcaption}>
            {title1 && (
              <Heading level="3" size="xSmall" theme={theme}>
                {link1 ? (
                  <Hyperlink {...link1} theme={theme}>
                    {title1}
                  </Hyperlink>
                ) : (
                  title1
                )}
              </Heading>
            )}
            {caption1 && (
              <div
                className={styles.caption}
                dangerouslySetInnerHTML={{ __html: caption1 }}
              />
            )}
          </figcaption>
        )}
      </Figure>
      <Figure
        caption={!title2 && !caption2 && imageVideo2.caption}
        className={styles.figure}
        hasCaptionBorder={false}
        theme={theme}
      >
        {link2 ? (
          <Hyperlink {...link2} className={styles.imageLinkWrapper}>
            <ImageVideoBlock2 />
          </Hyperlink>
        ) : (
          <ImageVideoBlock2 />
        )}
        {(title2 || caption2) && (
          <figcaption className={styles.figcaption}>
            {title2 && (
              <Heading level="3" size="xSmall" theme={theme}>
                {link2 ? (
                  <Hyperlink {...link2} theme={theme}>
                    {title2}
                  </Hyperlink>
                ) : (
                  title2
                )}
              </Heading>
            )}
            {caption2 && (
              <div
                className={styles.caption}
                dangerouslySetInnerHTML={{ __html: caption2 }}
              />
            )}
          </figcaption>
        )}
      </Figure>
    </div>
  );
};

const linkType = PropTypes.shape({
  text: PropTypes.string,
  url: PropTypes.string,
  hasTargetInNewWindow: PropTypes.bool,
  type: PropTypes.string,
});

DoubleImage.propTypes = {
  className: PropTypes.string,
  imageVideo1: PropTypes.oneOfType([
    PropTypes.shape({
      type: PropTypes.string,
      altText: PropTypes.string,
      caption: PropTypes.string,
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
    }),
    PropTypes.shape({
      type: PropTypes.string,
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
      poster: PropTypes.object,
    }),
  ]),
  imageVideo2: PropTypes.oneOfType([
    PropTypes.shape({
      type: PropTypes.string,
      altText: PropTypes.string,
      caption: PropTypes.string,
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
    }),
    PropTypes.shape({
      type: PropTypes.string,
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
      poster: PropTypes.object,
    }),
  ]),
  link1: linkType,
  link2: linkType,
  theme: PropTypes.oneOf(['dark', 'light']),
  title1: PropTypes.string,
  title2: PropTypes.string,
  caption1: PropTypes.string,
  caption2: PropTypes.string,
};

DoubleImage.defaultProps = {
  className: undefined,
  imageVideo1: undefined,
  imageVideo2: undefined,
  theme: 'dark',
  link1: undefined,
  link2: undefined,
  title1: undefined,
  title2: undefined,
  caption1: undefined,
  caption2: undefined,
};

export default DoubleImage;
