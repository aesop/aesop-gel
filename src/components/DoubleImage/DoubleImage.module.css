@import 'styles/index';

.base {
  @media (--viewport-md) {
    display: flex;
    justify-content: space-between;
  }
}

.figure {
  margin-bottom: calc(var(--layout-sm-spacing) * 2);

  @media (--viewport-md) {
    display: flex;
    flex: 0 0 calc(50% - var(--layout-md-spacing) / 4);
    flex-direction: column;
    margin-bottom: auto;
    margin-bottom: 0;
  }

  @media (--viewport-lg) {
    flex: 0 0 calc(50% - var(--layout-lg-spacing) / 4);
  }
}

.imageLinkWrapper {
  display: block;
}

.figcaption {
  padding-right: var(--layout-sm-spacing);

  .dark & {
    color: var(--color-dark-copy);
  }

  .light & {
    color: var(--color-light-copy);
  }

  @media (--viewport-md) {
    padding-right: var(--layout-md-spacing);
  }

  @media (--viewport-lg) {
    padding-right: var(--layout-lg-spacing);
  }
}

.caption {
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(14px);
  line-height: 1.6;
}
