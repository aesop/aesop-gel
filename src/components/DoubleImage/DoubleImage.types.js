import React from 'react';
import { ClassSetType, FigureType, ImageType } from '~/types';

export interface Props extends ClassSetType {
  id?: string | undefined;
  left: React.ReactElement<FigureType> | React.ReactElement<ImageType>;
  right: React.ReactElement<FigureType> | React.ReactElement<ImageType>;
}

export { Props as default }; // eslint-disable-line no-undef
