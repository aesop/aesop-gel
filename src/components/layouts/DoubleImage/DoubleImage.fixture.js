export default {
  id: 'double-image-careers-page',
  left: {
    figure: {
      caption: '',
      id: 'left-caption-id',
    },
    image: {
      altText: 'Milk Cleansing Campaign Aesop 2019',
      cta: {
        id: 'milk-cleansing,jpg',
        url: 'https://www.aesop.com/au/r/philosophy-to-products',
      },
      large:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
      medium:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
      small:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
    },
  },
  right: {
    figure: {
      caption: '',
      id: 'right-caption-id',
    },
    image: {
      altText: 'Milk Cleansing Campaign Aesop 2019',
      cta: {
        id: 'milk-cleansing,jpg',
        url: 'https://www.aesop.com/au/r/philosophy-to-products',
      },
      large:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
      medium:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
      small:
        'https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x',
    },
  },
};
