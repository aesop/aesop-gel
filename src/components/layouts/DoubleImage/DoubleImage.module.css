@import 'styles/variables/index';
@import 'styles/mixins/index';

.wrapper {
  padding: var(--layout-sm-spacing);

  @media (--viewport-md) {
    display: flex;
    justify-content: space-between;
    padding: var(--layout-md-spacing);
  }

  @media (--viewport-lg) {
    padding: var(--layout-lg-spacing) 80px;
  }
}

.item {
  width: 100%;
  flex-shrink: 0;

  @media (--viewport-sm-only) {
    &:not(:last-child) {
      margin-bottom: 35px;
    }
  }

  @media (--viewport-md) {
    width: calc(50% - 15px);
  }

  @media (--viewport-lg) {
    width: calc(50% - 10px);
  }

  figure {
    padding: 0;

    figcaption {
      padding: 20px 0 0 0;
      border: none;
    }

    p {
      color: var(--color-body-copy);
    }
  }
}
