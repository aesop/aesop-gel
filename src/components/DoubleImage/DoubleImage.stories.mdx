import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import DoubleImage from './DoubleImage';
import DoubleImageFixture from './DoubleImage.fixture';

<Meta title="Components/DoubleImage" component={DoubleImage} />

# DoubleImage

<Preview>
  <Story name="Base component">
    <DoubleImage
      mediaOne={{
        heading: knobs.text('Title 1', 'Title 1', 'Content'),
        link: knobs.object('Link 1', DoubleImageFixture.link, 'Links'),
        type: DoubleImageFixture.imageOne.type,
        altText: DoubleImageFixture.imageOne.altText,
        caption: DoubleImageFixture.imageOne.caption,
        large: DoubleImageFixture.imageOne.large,
        medium: DoubleImageFixture.imageOne.medium,
        small: DoubleImageFixture.imageOne.small,
      }}
      mediaTwo={{
        heading: knobs.text('Title 2', 'Title 2', 'Content'),
        link: knobs.object('Link 2', DoubleImageFixture.link, 'Links'),
        type: DoubleImageFixture.imageTwo.type,
        altText: DoubleImageFixture.imageTwo.altText,
        caption: DoubleImageFixture.imageTwo.caption,
        large: DoubleImageFixture.imageTwo.large,
        medium: DoubleImageFixture.imageTwo.medium,
        small: DoubleImageFixture.imageTwo.small,
      }}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Preview>

## Variations

### Video in slots

<Preview>
  <Story name="Contains videos">
    <DoubleImage
      mediaOne={{
        heading: knobs.text('Title 1', 'Title 1', 'Content'),
        link: knobs.object('Link 1', DoubleImageFixture.link, 'Links'),
        type: DoubleImageFixture.video.type,
        caption: DoubleImageFixture.caption,
        poster: DoubleImageFixture.video.poster,
        large: DoubleImageFixture.video.large,
        medium: DoubleImageFixture.video.medium,
        small: DoubleImageFixture.video.small,
      }}
      mediaTwo={{
        heading: knobs.text('Title 2', 'Title 2', 'Content'),
        link: knobs.object('Link 2', DoubleImageFixture.link, 'Links'),
        type: DoubleImageFixture.video.type,
        caption: DoubleImageFixture.caption,
        poster: DoubleImageFixture.video.poster,
        large: DoubleImageFixture.video.large,
        medium: DoubleImageFixture.video.medium,
        small: DoubleImageFixture.video.small,
      }}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Preview>

## API

<Props of={DoubleImage} />
