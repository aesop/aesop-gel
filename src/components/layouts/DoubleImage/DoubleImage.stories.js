import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Figure from '~/components/elements/Figure';
import Image from '~/components/elements/Image';
import DoubleImage from './DoubleImage';
import DoubleImageFixture from './DoubleImage.fixture';

storiesOf('Layouts.DoubleImage', module).add('Base component', () => (
  <DoubleImage
    id={text('id', DoubleImageFixture.id)}
    left={
      <Figure
        caption={text('caption', DoubleImageFixture.left.figure.caption)}
        id={text('id', DoubleImageFixture.left.figure.id)}
      >
        <Image
          altText={text('altText', DoubleImageFixture.left.image.altText)}
          cta={{
            id: text('id', DoubleImageFixture.left.image.cta.url),
            url: text('url', DoubleImageFixture.left.image.cta.url),
          }}
          large={text('large', DoubleImageFixture.left.image.large)}
          medium={text('medium', DoubleImageFixture.left.image.medium)}
          small={text('small', DoubleImageFixture.left.image.small)}
        />
      </Figure>
    }
    right={
      <Figure
        caption={text('caption', DoubleImageFixture.right.figure.caption)}
        id={text('id', DoubleImageFixture.right.figure.id)}
      >
        <Image
          altText={text('altText', DoubleImageFixture.right.image.altText)}
          cta={{
            id: text('id', DoubleImageFixture.right.image.cta.url),
            url: text('url', DoubleImageFixture.right.image.cta.url),
          }}
          large={text('large', DoubleImageFixture.right.image.large)}
          medium={text('medium', DoubleImageFixture.right.image.medium)}
          small={text('small', DoubleImageFixture.right.image.small)}
        />
      </Figure>
    }
  />
));
