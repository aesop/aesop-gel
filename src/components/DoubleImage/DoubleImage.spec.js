import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DoubleImage from './DoubleImage';
import DoubleImageFixture from './DoubleImage.fixture';

configure({ adapter: new Adapter() });

describe('<DoubleImage />', () => {
  it('should be defined', () => {
    expect(DoubleImage).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <DoubleImage
          caption1={<p>{DoubleImageFixture.caption}</p>}
          caption2={<p>{DoubleImageFixture.caption}</p>}
          imageVideo1={{
            altText: DoubleImageFixture.imageOne.altText,
            caption: DoubleImageFixture.imageOne.caption,
            large: DoubleImageFixture.imageOne.large,
            medium: DoubleImageFixture.imageOne.medium,
            small: DoubleImageFixture.imageOne.small,
          }}
          imageVideo2={{
            altText: DoubleImageFixture.imageTwo.altText,
            caption: DoubleImageFixture.imageTwo.caption,
            large: DoubleImageFixture.imageTwo.large,
            medium: DoubleImageFixture.imageTwo.medium,
            small: DoubleImageFixture.imageTwo.small,
          }}
          title1={DoubleImageFixture.title}
          title2={DoubleImageFixture.title}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
