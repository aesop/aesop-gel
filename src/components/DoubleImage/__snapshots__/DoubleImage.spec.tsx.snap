// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DoubleImage /> renders base component correctly 1`] = `
<div
  className="base"
  id="double-image-careers-page"
>
  <div
    className="wrapper"
  >
    <div
      className="item"
    >
      <figure
        className="base"
        id="left-caption-id"
      >
        <a
          className="base"
          href="https://www.aesop.com/au/r/philosophy-to-products"
          title="Link to https://www.aesop.com/au/r/philosophy-to-products"
        >
          <picture
            className="picture base"
          >
            <source
              media="(min-width: 1025px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <source
              media="(min-width: 640px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <source
              media="(min-width: 0px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <img
              alt="Milk Cleansing Campaign Aesop 2019"
            />
          </picture>
        </a>
        
      </figure>
    </div>
    <div
      className="item"
    >
      <figure
        className="base"
        id="right-caption-id"
      >
        <a
          className="base"
          href="https://www.aesop.com/au/r/philosophy-to-products"
          title="Link to https://www.aesop.com/au/r/philosophy-to-products"
        >
          <picture
            className="picture base"
          >
            <source
              media="(min-width: 1025px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <source
              media="(min-width: 640px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <source
              media="(min-width: 0px)"
              srcSet="https://www.aesop.com/medias/Aesop-AU-Store-Paddington-II-Hero-Bleed-Desktop-2880x1620px.jpg?context=bWFzdGVyfGltYWdlc3wxMDA1NjIzfGltYWdlL2pwZWd8aW1hZ2VzL2hkZC9oZDQvOTAyMDgzMzk4ODYzOC5qcGd8NjNkZjYzZTBkYzQ3OGEyMGYzMmQwOTI5YTMwZWE0ZTk2ZjE0MTNlYjY2MTk1MzRkZGYxMWRjMmMxMGY4NTRmMQ 2x"
            />
            <img
              alt="Milk Cleansing Campaign Aesop 2019"
            />
          </picture>
        </a>
        
      </figure>
    </div>
  </div>
</div>
`;
